evap/results/tools.py

```diff
@@ -179,12 +185,17 @@ def calculate_average_distribution(course):
 
     course_results = grouped_results.pop(None, [])
 
-    average_contributor_distribution = avg_distribution([
-        avg_distribution(
-            [average_grade_questions_distribution(results), average_non_grade_rating_questions_distribution(results)],
-            [settings.CONTRIBUTOR_GRADE_QUESTIONS_WEIGHT, settings.CONTRIBUTOR_NON_GRADE_RATING_QUESTIONS_WEIGHT]
-        ) for results in grouped_results.values()
-    ])
+    average_contributor_distribution = avg_distribution(
+        [
+            avg_distribution(
+                [average_grade_questions_distribution(contributor_results), average_non_grade_rating_questions_distribution(contributor_results)],
+                [settings.CONTRIBUTOR_GRADE_QUESTIONS_WEIGHT, settings.CONTRIBUTOR_NON_GRADE_RATING_QUESTIONS_WEIGHT]
+            ) for contributor_results in grouped_results.values()
+        ],
+        [
+            sum(sum(result.counts) for result in contributor_results if result.question.is_rating_question) for contributor_results in grouped_results.values()
```

Comment:
this seems to be the wrong order of the loops. please find out why this got through the tests.

Reply:
i did not properly recognize the parentheses, this is fine as it is.